From 0560b8e11ed2b53f9ef64d4c78fc51ea57348b80 Mon Sep 17 00:00:00 2001
From: Jakub Tyszkowski <jakub.tyszkowski@tieto.com>
Date: Tue, 2 Dec 2014 11:20:03 +0100
Subject: [PATCH 07/11] selinux: Add domain transition from logwrapper for
 bluetooth services

After logwrapper was provided with its own security domain we can set
proper domain transition rules and run bluetooth services without the
need of permissive security mode.

Change-Id: I8853367eeced947cf80c68a7ff6be494411b8d9c
---
 bluetoothd.te       |  8 +++++---
 bluetoothd_snoop.te | 10 +++++++---
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/bluetoothd.te b/bluetoothd.te
index 757dbda..4f229fa 100644
--- a/bluetoothd.te
+++ b/bluetoothd.te
@@ -2,12 +2,14 @@ type bluetoothd, domain;
 type bluetoothd_exec, exec_type, file_type;
 type bluetoothd_main_exec, exec_type, file_type;
 
-# FIXME: this is still here because of logwrapper
-permissive bluetoothd;
-
 # Start bluetoothd from init
 init_daemon_domain(bluetoothd)
 
+# logwrapper domain transition
+domain_auto_trans(logwrapper, bluetoothd_main_exec, bluetoothd)
+domain_auto_trans(logwrapper, bluetoothd_exec, bluetoothd)
+allow bluetoothd logwrapper:fd use;
+
 # Data file accesses
 allow bluetoothd bluetooth_data_file:dir w_dir_perms;
 allow bluetoothd bluetooth_data_file:notdevfile_class_set create_file_perms;
diff --git a/bluetoothd_snoop.te b/bluetoothd_snoop.te
index 626b583..e082341 100644
--- a/bluetoothd_snoop.te
+++ b/bluetoothd_snoop.te
@@ -1,12 +1,16 @@
 type bluetoothd_snoop, domain;
 type bluetoothd_snoop_exec, exec_type, file_type;
 
-# FIXME: this is still here because of logwrapper
-permissive bluetoothd;
-
 # Start bluetoothd_snoop from init
 init_daemon_domain(bluetoothd_snoop)
 
+# logwrapper domain transition
+domain_auto_trans(logwrapper, bluetoothd_snoop_exec, bluetoothd_snoop)
+allow bluetoothd_snoop logwrapper:fd use;
+
+# needs /system/bin/log access
+allow bluetoothd_snoop devpts:chr_file rw_file_perms;
+
 # directory search and read caps
 allow bluetoothd_snoop self:capability dac_read_search;
 # use raw and packet sockets caps
-- 
2.9.3

